### DISCUSSION TITLE
nspanel pro 3.4 ota

### DISCUSSION THREAD

**Discussion by ahhl** [seaky/nspanel_pro_tools_apk#134] [Q&A]
hi.
is it possible to share ota 3.4?
since i rooted my nspanel, this page have been life saver for me. i am now on. 3.0. hopefully i can update it to 3.4

**Comment by seaky** [comment]
There is no 3.4 OTA for nspanel86p, only nspanel120p has a new 3.4 ota fw.

But there is a 3.4 APK you can download here:
https://global-otadl2bsy.coolkit.cc/nspanel-pro/apk/65/app3.4.0.apk

**Comment by LenirSantiago** [comment]
3.5 is out... do you have the download link?

**Comment by seaky** [comment]
**nspanel86p:**
**apk**: https://global-otadl2bsy.coolkit.cc/nspanel-pro/apk/69/app3.5.0.apk
**fw**:  
- 3.0.0 - https://global-otadl2bsy.coolkit.cc/nspanel-pro/rom/27/CoolKit_Sonoff_480P_20240306_3.0.0-ota.zip
- 3.5.0 - https://global-otadl2bsy.coolkit.cc/nspanel-pro/rom-diff/31/CK_3.0.0_3.5.0-diff.zip

> [!WARNING]
> Before fw 3.5.0 you must update your device to fw 3.0.0! If you miss it you will permanently damage your device!
> Never upgrade firmware version 3.5 or later to a version below 3.0!

**nspanel120p:**
**apk**: https://global-otadl2bsy.coolkit.cc/nspanel-pro-ver120/apk/7/228V3.5.0.apk
**fw**: https://global-otadl2bsy.coolkit.cc/nspanel-pro-ver120/rom/7/SN_3326S_750X1334_4lan_龙腾_V3.5.0_20240913-ota.zip

> [!IMPORTANT]
> If you are an android device user, I definitely do not recommend updating the OS/FW you wont get any benefit! Zigbee related features are available without upgrade. The base OS version wont change because this device is designed to Android Oreo.
> Never upgrade firmware version 3.5 or later to a version below 3.0!

**Reply by eerke2021** [reply]
if anyone else does this. 3.5.0 was scary. Spent ages on firmware update screen. Thought it was bricked. 3.0 was longer to push but updated quicker. 3.5 has local control. Now need to figure out how to test.

incidentally "Never upgrade fw 3.5.0 to fw under 3.0.0!" should be 'from' instead of 'to', or I misunderstood it.

**Reply by seaky** [reply]
> if anyone else does this. 3.5.0 was scary. Spent ages on firmware update screen. Thought it was bricked. 3.0 was longer to push but updated quicker. 3.5 has local control. Now need to figure out how to test.

From firmware version 3.5 onward, all firmware updates are incremental, meaning they only contain changes relative to the previous version.
Since its smaller, it should be much faster...

> 
> 
> incidentally "Never upgrade fw 3.5.0 to fw under 3.0.0!" should be 'from' instead of 'to', or I misunderstood it.

Thx I fixed the typo.

**Reply by Wuehli** [reply]
> if anyone else does this. 3.5.0 was scary. Spent ages on firmware update screen. Thought it was bricked. 3.0 was longer to push but updated quicker. 3.5 has local control. 

Thanks for the hint!

**Comment by lukatmyshu** [comment]
I've been trying to enable developer mode in a newly installed 3.5.0 NSPanelPro and even after tapping the deviceID multiple times I don't see any agreement coming up.  Is there an alternate way or do I need to try using a USB OTG cable.

**Reply by lukatmyshu** [reply]
never mind - I realized you have to tap in the ewelink app (not in the settings list)

**Comment by masterx1981** [comment]
Sorry, there are advantages upgrading npanel pro (not 120) to 3.5?

**Reply by eerke2021** [reply]
The change log for 3.5 seems like they have added Local Mode to the eWeLink front end, but I think if you use Home Assistant the Sonoff Integration there used some local mode anyway, so if you used the NSPP to control Home Assistant I'm not sure it added anything. Also I haven't tested it yet. In the future the "paral-sync" is going to be bi-directional between iHost and NS Panel Pro. That may require a firmware update but I doubt it'll be 3.5. I wouldn't have bothered if I didn't use their own front end occasionally and am not waiting for the Paral-sync (iHost is not on home assistant and is fully local, but does have matter bridging so some stuff is sent that way).

**Comment by ahhl** [comment]
thanks seaky
i forgotten how to update the FW, because i still novice
can i update the OTA thru tcpip and what the steps and command to update the firmware and 
do i install 3.5.apk first or the fw first?

edited
ok. found the way to flash using your root tool

thanks

edit
i got stuck on how to transfer file to the device over tcpip.
any help. i use adb sideload filename, i got device not found after connected thru tcpip

**Reply by houtacheng** [reply]
eg:
adb push 1.mp3 /sdcard/mp3/

**Comment by houtacheng** [comment]
I was pleasantly surprised to find that after manually flashing version 3.5 and installing the 3.5 apk, the firmware version 3.7 can be automatically detected.
<img width="332" alt="image" src="https://github.com/user-attachments/assets/b5887211-b4a4-4353-a2f7-bfe06138a7ff" />
<img width="405" alt="image" src="https://github.com/user-attachments/assets/7f93721e-acbb-4ae7-a8d6-62eb9a34077e" />
<img width="562" alt="image" src="https://github.com/user-attachments/assets/c2d0fe68-0a14-4e46-b838-e85aa0137ce8" />
https://forum.ewelink.cc/t/nspanel-pro-v3-7-0-web-shortcut-for-easily-access-local-page/46743
<img width="561" alt="image" src="https://github.com/user-attachments/assets/a521d879-c4f3-4c9f-babc-59001dcc2e77" />

**Reply by eerke2021** [reply]
I installed 3.5 a while ago and don't get that option on the device. Had assumed the APK was just a repackaged version of the firmware. Not sure I ever installed it also. Hard to recall what exactly I did now. Pretty sure I just picked the firmware to where the root tool said to place it. Now I've set it to automatically launch the Sonoff UI, in the hope the Matter integration would work better, and I need to re-read how to go back.

**Reply by houtacheng** [reply]
nspanel86p：
apk：https://global-otadl2bsy.coolkit.cc/nspanel-pro/apk/69/app3.5.0.apk

**Reply by houtacheng** [reply]
https://github.com/seaky/nspanel_pro_roottool_apk/issues/1#issuecomment-2358450356

**Reply by houtacheng** [reply]
3.7 firmware
https://github.com/seaky/nspanel_pro_roottool_apk/issues/1#issuecomment-2553473010

**Reply by eerke2021** [reply]
Thanks. I still need to figure out how to get back to not auto launching the Sonoff launcher in order to access the NS Proroot Tool  it I have saved the file anyway for when I figure this out.

**Reply by houtacheng** [reply]
按下首頁按鈕：
adb shell 輸入 keyevent 3

按下電源按鈕：
adb shell 輸入 keyevent 26

顯示通知：
adb shell cmd statusbar Expand-notifications

**Comment by eerke2021** [comment]
Thanks. I tried the first two commands but nothing happened, so much I remotely rebooted to check it was working. I will try the third command. On 25 Dec 2024, at 16:05, houtacheng ***@***.***> wrote:﻿
按下首頁按鈕：
adb shell 輸入 keyevent 3
按下電源按鈕：
adb shell 輸入 keyevent 26
顯示通知：
adb shell cmd statusbar Expand-notifications

**Comment by DJBenson** [comment]
Is it possible to use the new feature of 3.7.0 without updating the entire firmware? I assume the feature is in the app package but is that true and will it install on firmware 2.5.0?

If so, does anybody have the 3.7.0 apks?

**Reply by eerke2021** [reply]
My understanding is that you must first update to 3.0 because at that point the update procedure changed and then there are different updates depending on where you start and end. For example 3.5 to 3.7 is not the same as 3.0 to 3.7. In your case you might want to see if just going to 3.0 has all the features you want.

**Reply by DJBenson** [reply]
The "native" Home Assistant feature was only introduced in 3.7.0 though. I'm just wondering if I take the plunge and just upgrade to 3.7.1 because at the moment my unit is little more than a brick - I've never been able to get usable performance out of the webview presentation of Home Assistant.

**Reply by eerke2021** [reply]
I think it's just a shortcut to the webpage. Haven't had a thorough look at it though. 

I'd try installing the eWeLink palette in Node Red on the Home Assistant and then creating virtual devices that are mapped to the Home Assistant devices you want to use from the NSP. Only works for a small number of devices though as each takes some time to map. Uses the native designed buttons on the NSP to control the HA though, so should be as quick as you want.

**Comment by seaky** [comment]
https://global-otadl2bsy.coolkit.cc/nspanel-pro/apk/79/app3.8.0.apk
